pkg/data/cloud/platform.go

```diff
@@ -0,0 +1,63 @@
+package platform
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// Platform type represents specific Platform types and how they map to their respective platforms.
+type Platform struct {
+	// names holds 2 unique lowercase names of the Platform (e.g., "aws"). We use a fixed-
+	// size array so that this struct remains comparable. Any of the 2 values can be used to refer
+	// to this specific Platform via Platform.PlatformByName(), but only the first (element
+	// 0) element will be the "preferred name" returned by Platform.String()
+	names [2]string
+}
+
+var (
+	AWSClassic = Platform{
+		names: [2]string{"aws", "aws-classic"},
+	}
+	AWSHCP = Platform{
+		names: [2]string{"hostedcluster", "aws-hcp"},
+	}
+	GCPClassic = Platform{
+		names: [2]string{"gcp", "gcp-classic"},
+	}
+)
+
+// String returns the "preferred name" of the Platform
+func (plat Platform) String() string {
+	return plat.names[0]
+}
+
+// PlatformByName returns a Platform supported by the verifier if the given name
+// matches any known common names for a supported Platform. It returns an empty/invalid
+// platform if the provided name isn't supported
+func PlatformByName(name string) (Platform, error) {
```

Comment:
Go convention would have us name this `ByName`, so it's called like `platform.ByName` - I could go either way here personally.

Reply:
I'm cool with either option as well. I'll leave it up to @dakotalongRH

Reply:
Agreed, changing to platform.ByName